Update crates edition and clippy

BlackDex opened the pull request:
# General

Updated several aspects this crate.


## Update crates
    
Updated crates to the latest version.
Modified the base64 code to match the new version.

## Apply clippy fixes

Applied clippy fixes where sugested.

## Updated examples

- Synced code between all examples so it can be compared better.
- Made all example runnable by requesting a token and let them be
  configured via env variables.

## Update Dockerfile and Readme

- Update the Dockerfile to use the stable version of rust instead of a fixed version.
- Use `apk --no-cache` instead of running `apk update`
- Added an `EXAMPLE` build-arg to build a different example instead of
  the default `otp` example.
- Some other small changes

Also added a static build Dockerfile. This can build a static version of
the `otp` examples and can be extracted and used on almost any Linux
based system, and it is a nice test.

## Add GitHub Actions Workflow

Add a GitHub actions workflow to run checks and build all the examples
to validate if at least building the code works.

BlackDex committed:
Update crates

Updated crates to the latest version.
Modified the base64 code to match the new version.

BlackDex committed:
Apply clippy fixes

Applied clippy fixes where sugested.

BlackDex committed:
Updated examples

- Synced code between all examples so it can be compared better.
- Made all example runnable by requesting a token and let them be
  configured via env variables.

BlackDex committed:
Update Dockerfile and Readme

- Update the Dockerfile to use the stable version of rust instead of a fixed version.
- Use `apk --no-cache` instead of running `apk update`
- Added an `EXAMPLE` build-arg to build a different example instead of
  the default `otp` example.
- Some other small changes

Also added a static build Dockerfile. This can build a static version of
the `otp` examples and can be extracted and used on almost any Linux
based system, and it is a nice test.

BlackDex committed:
Add GitHub Actions Workflow

Add a GitHub actions workflow to run checks and build all the examples
to validate if at least building the code works.

BlackDex commented:
Also, to see a GitHub Actions run, you can check my fork: https://github.com/BlackDex/yubico-rs/actions/runs/8544333651

BlackDex commented:
@pierre-l any chance for a new release including this PR?

pierre-l commented:
@BlackDex I don't think I even have a crates.io account :/

@wisespace-io FYI

Conni2461 commented:
we would also be very interested in seeing this merged :) thanks for the work

friendly reminder @wisespace-io

BlackDex commented:
I tested this with the latest nightly and all the deps updated to the latest version and it still works just fine.
Merging this would solve a lot of duplicate dependencies :)

BlackDex commented:
Friendly request to @wisespace-io / @flavio-greenfact

BlackDex committed:
Update and release yubico_ng

* Upgrade to `tokio` 1.44, `rand` 0.9
* Renamed to yubico_ng and published crate
* Made edition 2024 compatible
* Added several clippy/rust lints and fixed those
* Fixed a panic if the `YK_API_HOST` was invalid
* Use only the main api server, the others are deprecated
* Run cargo fmt
* Updated GHA to use hashes and run/fix zizmor

Signed-off-by: BlackDex <black.dex@gmail.com>

BlackDex committed:
Set MSRV based upon 'cargo msrv find'

Signed-off-by: BlackDex <black.dex@gmail.com>

BlackDex commented:
For everyone who wants to use these changes and more, i have created a new crate very creatively called `yubico_ng`.
https://crates.io/crates/yubico_ng

I have updated it to the latest crates currently available and made it Edition 2024 compatible including some clippy fixes and fixed a panic.

Conni2461 commented:
thanks for forking :D
